src/components/VaultRow/lib/VaultApy.tsx

```diff
@@ -0,0 +1,27 @@
+import { FC } from "react"
+
+import { CompounderVaultProps, ConcentratorVaultProps } from "@/lib/types"
+import { useIsCompounderProduct } from "@/hooks/useVaultProduct"
+
+import { CompounderVaultApy } from "@/components/Compounder"
+import { ConcentratorVaultApy } from "@/components/Concentrator"
+import { VaultRowPropsWithProduct } from "@/components/VaultRow"
+
+export const VaultApy: FC<VaultRowPropsWithProduct> = ({
```

Comment:
I think for simplicity's sake, unless there is some sort of data fetching happening inside here, I'd rather just see this logic inlined in the VaultRow component (in the same way the deposit/withdraw logic is inlined). 

---

If you really want to clean up the ternaries more, you should create a generic component that will display one of two components based on `productType`:

```tsx
type ProductTypeComponentProps = {
  compounder: JSX.Element
  concentrator: JSX.Element
  productType: ProductType
}

const ProductTypeComponent = FC<ProductTypeComponentProps> = ({ compounder, concentrator, productType }) => {
  return <>{productType === "compounder" ? compounder : concentrator}</>
}

// usage
<ProductTypeComponent
  compounder={<CompounderVaultApy {...props} />}
  concentrator={<ConcentratorVaultApy {...props />}
  productType
/>
```

Not really necessary though and my code above is probably going to have issues with the prop types.

Reply:
Interesting point, I will dig it more.

Reply:
Will iterate on it later.